cli/internal/config/config_file_test.go

```diff
@@ -9,6 +9,51 @@ import (
 	"github.com/vercel/turborepo/cli/internal/fs"
 )
 
+func Test_UserConfigPath(t *testing.T) {
+	fsys := afero.NewMemMapFs()
+
+	tests := []struct {
+		name string
+		fsys afero.Fs
+	}{
+		{
+			name: "get doesn't create a path",
+			fsys: fsys,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			// XDG is not filesystem aware. Clean up first.
+			path, _ := createUserConfigPath(fsys)
+			err := os.Remove(path.Dir().ToString())
+
+			if err != nil {
+				t.Errorf("failed to clean up first: %v", err)
+			}
+
+			gotConfigPath, getErr := getUserConfigPath(tt.fsys)
+			if gotConfigPath == "" || getErr == nil {
+				gotConfigPath = path
+			}
+			gotConfigDir := gotConfigPath.Dir()
+			getCheck, _ := os.Stat(gotConfigDir.ToString())
```

Comment:
This is maybe a good candidate for `AbsolutePath.DirExists`, a sibling to `AbsolutePath.FileExists`.

Reply:
`DirExists` as an API is a bit confusing. Should it call `Dir` on the path first? Or should it process the entire path? `FileExists` suffers a similar problem in reverse: given a path that exists and is a directory, `FileExists` will return `false`. Expected behavior? Maybe. Confusing? Definitely.

Further, what should it do for symlinks? For example, without checking, I couldn't tell you whether `AbsolutePath.FileExists()` is going to follow a symlink or tell me about the link itself. (Turns out, our current implementation for `FileExists` calls `Lstat`, and returns `!FileInfo.IsDir()`.)

I suspect we might want to align toward `Mode().IsDir()` and `Mode().IsRegular()` which at least implies that we're investigating the `FileInfo`s of the current full `AbsolutePath`. But then there is the question of "how much of the [Mode API](https://cs.opensource.google/go/go/+/refs/tags/go1.18.3:src/os/types.go;l=35-56) do we want to provide sugar for?" And, most importantly, do we want to distinguish between reading symlinks or not?

We could do something like this, but it gets unwieldy:
```
AbsolutePath.IsDir()
AbsolutePath.IsRegular()
AbsolutePath.IsLink()
AbsolutePath.FollowLinksIsDir()
AbsolutePath.FollowLinksIsRegular()
```

In a "show your work" world I think I might prefer something like:
```
// Always returns a chainable call.
// If it errors, the error gets propagated to the last call.
// Intermediate chainable nodes always return an `APFileInfo`.
// Leaf chainable nodes either return the first `error` or a `bool`
// Errors are coerced to false.
func (ap AbsolutePath) Stat() APFileInfo {}
func (ap AbsolutePath) Lstat() APFileInfo {}

func (fi *APFileInfo) IsDir() bool {}

somePath.Stat().IsDir()
somePath.Lstat().IsDir()
```

Too much design work for a quick usage in a test; I'm going to leave this as is for now and we can circle back to discuss what this API should look like.